internal/kibana/fleet.go

```diff
@@ -24,34 +26,42 @@ type FleetConfig struct {
 	KibanaURI                string
 	FleetServerPort          int
 	FleetServerURI           string
+	FleetServerScheme        string
 }
 
 // NewFleetConfig builds a new configuration for the fleet agent, defaulting fleet-server host, ES credentials, URI and port.
 func NewFleetConfig(token string) (*FleetConfig, error) {
 	fleetServer := shell.GetEnv("FLEET_URL", "fleet-server")
 	fleetPort := 8220
+	fleetServerScheme := "http"
 	if fleetServer != "fleet-server" {
+		fleetServer = utils.RemoveQuotes(fleetServer)
 		u, err := url.Parse(fleetServer)
 		if err != nil {
-			log.Fatal("Could not parse FLEET_URL")
+			log.WithField("error", err).Fatal("Could not parse FLEET_URL")
 		}
 		host, port, err := net.SplitHostPort(u.Host)
 		if err != nil {
 			log.Fatal("Could not determine fleet port from FLEET_URL")
 		}
 		fleetPort, _ = strconv.Atoi(port)
 		fleetServer = host
+		fleetServerScheme = u.Scheme
 	}
 
+	esEndpoint := elasticsearch.GetElasticSearchEndpoint()
+	kbEndpoint := GetKibanaEndpoint()
+
 	cfg := &FleetConfig{
 		EnrollmentToken:          token,
-		ElasticsearchCredentials: "elastic:changeme",
-		ElasticsearchPort:        9200,
-		ElasticsearchURI:         "elasticsearch",
-		KibanaPort:               5601,
-		KibanaURI:                "kibana",
+		ElasticsearchCredentials: fmt.Sprintf("elastic:%s", shell.GetEnv("ELASTICSEARCH_PASSWORD", "changeme")),
```

Comment:
Should we embed the credentials in the Elasticsearch Endpoint struct, to avoid reading the password multiple times?